core/src/main/java/io/grpc/PickFirstBalancerFactory.java

```diff
@@ -46,138 +46,122 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1771")
 public final class PickFirstBalancerFactory extends LoadBalancer.Factory {
 
-  private static final PickFirstBalancerFactory instance = new PickFirstBalancerFactory();
+  private static final PickFirstBalancerFactory INSTANCE = new PickFirstBalancerFactory();
 
   private PickFirstBalancerFactory() {
   }
 
   public static PickFirstBalancerFactory getInstance() {
-    return instance;
+    return INSTANCE;
   }
 
   @Override
-  public <T> LoadBalancer<T> newLoadBalancer(String serviceName, TransportManager<T> tm) {
-    return new PickFirstBalancer<T>(tm);
+  public LoadBalancer newLoadBalancer(LoadBalancer.Helper helper) {
+    return new PickFirstBalancer(helper);
   }
 
-  private static class PickFirstBalancer<T> extends LoadBalancer<T> {
-    private static final Status SHUTDOWN_STATUS =
-        Status.UNAVAILABLE.augmentDescription("PickFirstBalancer has shut down");
-
-    private final Object lock = new Object();
-
-    /** "lock" must be held when mutating. */
-    private volatile EquivalentAddressGroup addresses;
-    @GuardedBy("lock")
-    private InterimTransport<T> interimTransport;
-    @GuardedBy("lock")
-    private Status nameResolutionError;
-    @GuardedBy("lock")
-    private boolean closed;
-
-    private final TransportManager<T> tm;
+  @VisibleForTesting
+  static class PickFirstBalancer extends LoadBalancer {
+    private final Helper helper;
+    private Subchannel subchannel;
 
-    private PickFirstBalancer(TransportManager<T> tm) {
-      this.tm = tm;
+    public PickFirstBalancer(Helper helper) {
+      this.helper = helper;
```

Comment:
checkNotNull for public methods.  Also, reduce visibility to match class visibility